lib/active_merchant/billing/gateways/beanstream.rb

```diff
@@ -65,147 +66,25 @@ module Billing #:nodoc:
+      include Empty
 
-      def authorize(money, source, options = {})
-        post = {}
-        add_amount(post, money)
-        add_invoice(post, options)
-        add_source(post, source)
-        add_address(post, options)
-        add_transaction_type(post, :authorization)
-        add_customer_ip(post, options)
-        add_recurring_payment(post, options)
-        commit(post)
-      end
-
-      def purchase(money, source, options = {})
-        post = {}
-        add_amount(post, money)
-        add_invoice(post, options)
-        add_source(post, source)
-        add_address(post, options)
-        add_transaction_type(post, purchase_action(source))
-        add_customer_ip(post, options)
-        add_recurring_payment(post, options)
-        commit(post)
-      end
-
-      def void(authorization, options = {})
-        reference, amount, type = split_auth(authorization)
-
-        post = {}
-        add_reference(post, reference)
-        add_original_amount(post, amount)
-        add_transaction_type(post, void_action(type))
-        commit(post)
-      end
-
-      def verify(source, options={})
-        MultiResponse.run(:use_first_response) do |r|
-          r.process { authorize(100, source, options) }
-          r.process(:ignore_result) { void(r.authorization, options) }
-        end
-      end
-
-      def success?(response)
-        response[:trnApproved] == '1' || response[:responseCode] == '1'
-      end
-
-      def recurring(money, source, options = {})
-        ActiveMerchant.deprecated RECURRING_DEPRECATION_MESSAGE
-
-        post = {}
-        add_amount(post, money)
-        add_invoice(post, options)
-        add_credit_card(post, source)
-        add_address(post, options)
-        add_transaction_type(post, purchase_action(source))
-        add_recurring_type(post, options)
-        commit(post)
-      end
-
-      def update_recurring(amount, source, options = {})
-        ActiveMerchant.deprecated RECURRING_DEPRECATION_MESSAGE
-
-        post = {}
-        add_recurring_amount(post, amount)
-        add_recurring_invoice(post, options)
-        add_credit_card(post, source)
-        add_address(post, options)
-        add_recurring_operation_type(post, :update)
-        add_recurring_service(post, options)
-        recurring_commit(post)
-      end
+      def self.new(options = {})
+        # region: 0 - US & CA, 1 - AU & NZ
+        # I want to add this logic to judge the region from shipping & billing address
+        # but don't know if it is a correct way.
+        # I think this can be decided by shopify which will integrate with.
 
-      def cancel_recurring(options = {})
-        ActiveMerchant.deprecated RECURRING_DEPRECATION_MESSAGE
+        # [ options[:billing_address], options[:shipping_address] ].compact.each do |address|
+        #   next if empty?(address[:country])
+        #   unless ['AU', 'NZ'].include?(address[:country])
+        #     @region = 1
+        #   end
+        # end
```

Comment:
✂️ please remove these comments. To make the default explicit, please modify the following line to be 
```
region = options[:region] || 0
```

Reply:
an alternative to integers would be to have symbols represent the regions, example: `:pacific`, `:america`... This will make it easier to know what region is being used without having to refer back to this file